bookwyrm/models/bookwyrm_import_job.py

```diff
@@ -22,23 +33,133 @@ class BookwyrmImportJob(ParentJob):
 
     archive_file = FileField(null=True, blank=True)
     import_data = JSONField(null=True)
-    required = DjangoArrayField(CharField(max_length=50, blank=True), blank=True)
+    required = DjangoArrayField(
+        models.fields.CharField(max_length=50, blank=True), blank=True
+    )
+    retry = BooleanField(default=False)
+
+    def start_job(self):
+        """Start the job"""
+        start_import_task.delay(job_id=self.id)
+
+    @property
+    def book_tasks(self):
+        """How many import book tasks are there?"""
+        return UserImportBook.objects.filter(parent_job=self).all()
+
+    @property
+    def status_tasks(self):
+        """How many import status tasks are there?"""
+        return UserImportPost.objects.filter(parent_job=self).all()
+
+    @property
+    def relationship_tasks(self):
+        """How many import relationship tasks are there?"""
+        return UserRelationshipImport.objects.filter(parent_job=self).all()
+
+    @property
+    def item_count(self):
+        """How many total tasks are there?"""
+        return self.book_tasks.count() + self.status_tasks.count()
+
+    @property
+    def pending_item_count(self):
+        """How many tasks are incomplete?"""
+        status = BookwyrmImportJob.Status
+        book_tasks = self.book_tasks.filter(
+            status__in=[status.PENDING, status.ACTIVE]
+        ).count()
+
+        status_tasks = self.status_tasks.filter(
+            status__in=[status.PENDING, status.ACTIVE]
+        ).count()
+
+        relationship_tasks = self.relationship_tasks.filter(
+            status__in=[status.PENDING, status.ACTIVE]
+        ).count()
+
+        return book_tasks + status_tasks + relationship_tasks
+
+    @property
+    def percent_complete(self):
+        """How far along?"""
+        item_count = self.item_count
+        if not item_count:
+            return 0
+        return math.floor((item_count - self.pending_item_count) / item_count * 100)
+
+
+class UserImportBook(ChildJob):
+    """ChildJob to import each book.
+    Equivalent to ImportItem when importing a csv file of books"""
+
+    book = ForeignKey(models.Book, on_delete=SET_NULL, null=True, blank=True)
+    book_data = JSONField(null=False)
+    fail_reason = TextField(null=True)
```

Comment:
Right now, the fail reason is never showing up in the UI, even though it is being set correctly. That's because these classes inherit from a non-abstract class (ChildJob), so when they're called as that super-class, as they are in the `user_troubleshoot` view, they don't have the attributes that only exist in the child classes accessible. 

There are two ways to fix this; one is to make `fail_reason` a property of the parent class, `ChildJob` (I suspect this is the right solution for fail reason). If there's a reason not to, or for a field that shouldn't be present on all the child classes (like `status_type`), you have to make the parent model use the inheritance manager and then call `select_subclasses()` when loading all the children in the view. This is a workaround for what you're doing when you write `item.userimportpost`, for example. 

I know that's confusing, but all you have to do is add to `bookwyrm/models/job.py`:
```python3
from model_utils.managers import InheritanceManager
...

 class ChildJob(Job):
    ...
    objects = InheritanceManager() 
```

And then call `items = job.child_jobs.all().select_subclasses().order_by("task_id").filter(status="failed")` on line 27 of bookwyrm/views/imports/user_troubleshoot.py

Once you do that, you don't have to use `item.userimportpost.book`, you can just do `item.book` in the template, and fail reason will start working correctly. 

(Sorry does that make any sense?)